### ISSUE TITLE
Reloading configuration first adds hooks, then removes them

### ISSUE THREAD

**Issue by gdubicki** [adnanh/webhook#150]
**OS**: CentOS Linux release 7.1.1503 (Core)

**webhook version**: from https://github.com/adnanh/webhook/pull/143

When I edit hooks.yaml with vim then after saving it I see the following in webhook log:

```
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36 hooks file /opt/webhook/current/conf/hooks.yaml overwritten
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36 attempting to reload hooks from /opt/webhook/current/conf/hooks.yaml
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36 found 2 hook(s) in file
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36         loaded: merge-request
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36         loaded: monitoring
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36 hooks file /opt/webhook/current/conf/hooks.yaml removed, no longer watching this file for changes, removing hooks that were loaded from it
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36         removing: merge-request
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36         removing: monitoring
Jul 07 09:56:36 host.company.com webhook[16452]: [webhook] 2017/07/07 09:56:36 removed 2 hook(s) that were loaded from file /opt/webhook/current/conf/hooks.yaml
```

..and the hooks are in fact not loaded - requests to them result in 404s.

**Comment by adnanh** [comment]
This is the issue of the *nix signals and editor implementations, instead of sending MODIFIED signal when overwriting the existing file, some editors actually delete the file, and create a new one with the new contents (i.e. direct result of working in a temporary file, and then copying it over the original one).

Try using different editor, or if that is not an option, or does not help, you will unfortunately be unable to use the `-hotreload` flag. In that case, you can still use `kill -USR1 webhookpid` to reload the hooks without the webhook restart.

**Comment by gdubicki** [comment]
Thanks for a prompt answer! :)

I would add a warning in the docs if this feature is not compatible with so popular editors as vim then.

**Comment by adnanh** [comment]
Feel free to do it! :-)

**Comment by adnanh** [comment]
Fixed in v2.6.6 by #172
